sql/layers/layers.go

```diff
@@ -0,0 +1,95 @@
+package layers
+
+import (
+	"fmt"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/sql"
+)
+
+// Status of the layer.
+type Status int8
+
+func (s Status) String() string {
+	switch s {
+	case Latest:
+		return "latest"
+	case Processed:
+		return "processed"
+	case Applied:
+		return "applied"
+	default:
+		panic(fmt.Sprintf("unknown status %d", s))
+	}
+}
+
+const (
+	// Latest layer that was added to db.
+	Latest Status = iota
+	// Processed layer is either synced from peers or updated by hare.
+	Processed = iota
+	// Applied layer to the state.
+	Applied
+)
+
+// SetHareOutput for the layer to a block id.
+func SetHareOutput(db sql.Executor, lid types.LayerID, output types.BlockID) error {
+	if _, err := db.Exec(`insert into layers (id, hare_output) values (?1, ?2) 
+					on conflict(id) do update set hare_output=?2;`,
+		func(stmt *sql.Statement) {
+			stmt.BindInt64(1, int64(lid.Value))
+			stmt.BindBytes(2, output[:])
+		}, nil); err != nil {
+		return fmt.Errorf("set hare output %s: %w", lid, err)
+	}
+	return nil
+}
+
+// GetHareOutput for layer.
+func GetHareOutput(db sql.Executor, lid types.LayerID) (rst types.BlockID, err error) {
```

Comment:
i don't get the combo. "name returned value + returning explicitly"
i do prefer to not have named returned values.

Reply:
err is assigned inside the closure. 

i think this is normal and shorter instead of defining rst and err as

```
var (
     rst ...
     err error
)
```

Reply:
i understand the syntax, i just prefer the longer syntax because i've seen quite a few subtle bugs caused by named return variables in long methods.